pkg/querier/querier.go

```diff
@@ -359,6 +359,7 @@ func (q querier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Mat
 	sets := make(chan storage.SeriesSet, len(q.queriers))
 	for _, querier := range q.queriers {
 		go func(querier storage.Querier) {
+			// We should always select sorted here as we will need to merge the series
 			sets <- querier.Select(true, sp, matchers...)
 		}(querier)
```

Comment:
Probably only one querier has matched series. But since we don't know, we have to set it to sort anyway?

Reply:
On cortex we can have multiple queries that has matchs... As the same series can (and probably will) be both on store gateway and ingesters.

In those cases we query ingester to get the recent data and the store gateway to query the historical data for the same series.

Reply:
Make sense?

Reply:
Yeah, this LGTM